Packs/knowbe4Phisher/Integrations/knowbe4Phisher/knowbe4Phisher.py

```diff
@@ -0,0 +1,770 @@
+  phisherMessages(all: false, page: 1, per: 25, query: {}) {{
+    pagination {{
+      page
+      pages
+      per
+      totalCount
+    }}
+  }}
+}}
+"""
+
+FETCH_WITHOUT_EVENTS = """query {{
+  phisherMessages(all: false, page: 1, per: {}, query: {}) {{
+    nodes {{
+      actionStatus
+      attachments(status: UNKNOWN) {{
+        actualContentType
+        filename
+        md5
+        reportedContentType
+        s3Key
+        sha1
+        sha256
+        size
+        ssdeep
+        virustotal {{
+          permalink
+          positives
+          scanned
+          sha256
+        }}
+      }}
+      category
+      comments {{
+        body
+        createdAt
+      }}
+      events {{
+        causer
+        createdAt
+        eventType
+        id
+        triggerer
+      }}
+      from
+      id
+      links(status: UNKNOWN) {{
+        dispositions
+        firstSeen
+        id
+        lastSeen
+        scheme
+        target
+        url
+        virustotal {{
+          permalink
+          positives
+          scanned
+          sha256
+        }}
+      }}
+      phishmlReport {{
+        confidenceClean
+        confidenceSpam
+        confidenceThreat
+      }}
+      pipelineStatus
+      rawUrl
+      reportedBy
+      rules {{
+        createdAt
+        description
+        id
+        matchedCount
+        name
+        tags
+      }}
+      severity
+      subject
+      tags {{
+        name
+        type
+      }}
+    }}
+    pagination {{
+      page
+      pages
+      per
+      totalCount
+    }}
+  }}
+}}"""
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any XSOAR logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this  implementation, no special attributes defined
+    """
+
+    def __init__(self, base_url, verify, proxy, first_fetch_time, headers=None, max_fetch=None):
+        self.first_fetch_time = first_fetch_time
+        self.max_fetch = max_fetch
+
+        super().__init__(base_url=base_url, verify=verify, headers=headers, proxy=proxy)
+
+    def phisher_gql_request(self, payload: str):
+        return self._http_request(
+            method='POST',
+            data=payload
+        )
+
+
+''' HELPER FUNCTIONS '''
+
+
+def create_request(st: str):
+    """ Function that creates a valid request to use in http_request.
+
+    args:
+        st: A multiline string with a request in a human readable format
+
+    Returns:
+        A string that is formatted to suit the http_request function
+    """
+    format_desc = st.splitlines()
+    final = ""
+    for i in format_desc:
+        final = final + i + '\\n'
+    final = final[:-2]
+    return final
+
+
+def calculate_number_of_events(client: Client, query: str):
```

Comment:
Please add this to all functions |(some are missing this)
```suggestion
def calculate_number_of_events(client: Client, query: str) -> str:
```